akvo/rsr/tests/results_framework/test_cumulative_updates.py

```diff
@@ -0,0 +1,239 @@
+from datetime import date
+from decimal import Decimal
+from akvo.rsr.tests.base import BaseTestCase
+from akvo.rsr.tests.utils import ProjectFixtureBuilder
+from akvo.rsr.models.result.utils import PERCENTAGE_MEASURE
+
+
+class CumulativeTestMixin:
+    PERIOD_1_START = date(2020, 1, 1)
+    PERIOD_2_START = date(2021, 1, 1)
+    DISAGGREGATION_CATEGORY = 'Gender'
+    DISAGGREGATION_TYPE_1 = 'Male'
+    DISAGGREGATION_TYPE_2 = 'Female'
+
+    def setup_project(self, percentage=False, contributors=None):
+        builder = ProjectFixtureBuilder().with_disaggregations({
+            self.DISAGGREGATION_CATEGORY: [
+                self.DISAGGREGATION_TYPE_1,
+                self.DISAGGREGATION_TYPE_2,
+            ]
+        }).with_results([{
+            'title': 'Result #1',
+            'indicators': [{
+                'title': 'Indicator #1',
+                'measure': PERCENTAGE_MEASURE if percentage else '',
+                'cumulative': True,
+                'periods': [
+                    {'period_start': self.PERIOD_1_START, 'period_end': date(2020, 12, 31)},
+                    {'period_start': self.PERIOD_2_START, 'period_end': date(2021, 12, 31)}
+                ]
+            }]
+        }])
+        if contributors:
+            builder.with_contributors(contributors)
+        return builder.build()
+
+
+class SingleUserCumulativeUnitUpdatesTestCase(BaseTestCase, CumulativeTestMixin):
```

Comment:
Mixins should come first due to [class resolution being from left to right](https://docs.python.org/3/tutorial/classes.html#multiple-inheritance).

Reply:
Fixed